pkg/api/dataproxy.go

```diff
@@ -116,17 +119,43 @@ func ProxyDataSourceRequest(c *middleware.Context) {
 	}
 
 	if ds.Type == m.DS_ES {
-		if c.Req.Request.Method == "DELETE" {
```

Comment:
this code block that handles the ES data proxy needs to be broken out into separate component with tests, having all this ES code in side this function feels like a bad idea.